src/lib/components/CommunitySelectionModal/CommunitySelectionSearch.js

```diff
@@ -37,13 +35,27 @@ export class CommunitySelectionSearch extends Component {
 
     this.state = {
       selectedEndpoint: defaultEndpoint,
+      overriddenComponents: props.overriddenComponents,
     };
   }
 
+  componentDidMount() {
+    const { appName } = this.props;
```

Comment:
why was this moved inside the didMount? It most likely causes the components to call the render function twice

Reply:
It will render twice only if we pass cmps to override the existing one, but you are right we can move this logic to the constructor itself :+1: